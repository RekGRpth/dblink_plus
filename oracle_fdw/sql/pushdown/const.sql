\c contrib_regression_utf8

EXPLAIN SELECT id FROM number_numeric1 WHERE val = '1000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000';
SELECT id FROM number_numeric1 WHERE val = '1000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000';

EXPLAIN (COSTS FALSE) SELECT * FROM tstz9_tstz WHERE val < '2011-02-03 12:34:56.123457 Asia/Tokyo' AND val >= '2011-02-03 12:34:56.123456 Asia/Tokyo';
SELECT * FROM tstz9_tstz WHERE val >= '2011-02-03 12:34:56.123456 Asia/Tokyo' AND val < '2011-02-03 12:34:56.123457 Asia/Tokyo';
SELECT * FROM tstz9_tstz WHERE val = '2011-02-03 12:34:56.123456 Asia/Tokyo';
SELECT * FROM tstz9_tstz WHERE val = '2011-02-03 12:34:56.123457 Asia/Tokyo';

\q
EXPLAIN SELECT val_date1 FROM t_pushdown2 WHERE val_intervalym =
'-1 years -1 mons';
EXPLAIN SELECT val_date1 FROM t_pushdown2 WHERE val_intervalym =
'1 years -1 mons';
EXPLAIN SELECT val_date1 FROM t_pushdown2 WHERE val_intervalym =
'1 mons 1 day';
SELECT val_date1 FROM t_pushdown2 WHERE val_intervalym =
'-1 years -1 mons';

SET timezone = 'Japan';
EXPLAIN SELECT val_date1 FROM t_pushdown2 WHERE val_date1 = 'infinity';
SELECT val_date1 FROM t_pushdown2 WHERE val_date1 = 'infinity';
EXPLAIN SELECT val_date1 FROM t_pushdown2 WHERE val_date1 = '-infinity';
SELECT val_date1 FROM t_pushdown2 WHERE val_date1 = '-infinity';
EXPLAIN SELECT val_date1 FROM t_pushdown2 WHERE val_date1 = '2011-01-02 BC';
SELECT val_date1 FROM t_pushdown2 WHERE val_date1 = '2011-01-02 BC';
EXPLAIN SELECT val_date1 FROM t_pushdown2 WHERE val_date2 = '2011-01-02 BC 12:34:56';
SELECT val_date1 FROM t_pushdown2 WHERE val_date2 = '2011-01-02 BC 12:34:56';
EXPLAIN SELECT val_date1 FROM t_pushdown2 WHERE val_timestamp = '2011-01-02 BC 12:34:56.123456';
SELECT val_date1 FROM t_pushdown2 WHERE val_timestamp = '2011-01-02 BC 12:34:56.123456';
EXPLAIN SELECT val_date1 FROM t_pushdown2 WHERE val_timestamptz = '2011-01-02 12:34:56.123456 Japan BC';
SELECT val_date1 FROM t_pushdown2 WHERE val_timestamptz = '2011-01-02 12:34:56.123456 Japan BC';
EXPLAIN SELECT val_date1 FROM t_pushdown2 WHERE val_timestampltz = '2011-01-02 BC 12:34:56.123456 Japan';
SELECT val_date1 FROM t_pushdown2 WHERE val_timestampltz = '2011-01-02 BC 12:34:56.123456 Japan';
EXPLAIN SELECT val_date1 FROM t_pushdown2 WHERE val_intervalym = '-177999999 year -11 month';
SELECT val_date1 FROM t_pushdown2 WHERE val_intervalym = '-177999999 year -11 month';
EXPLAIN SELECT val_date1 FROM t_pushdown2 WHERE val_intervalds = '-999999999 day -23:59:59.999999';
SELECT val_date1 FROM t_pushdown2 WHERE val_intervalds = '-999999999 day -23:59:59.999999';

EXPLAIN SELECT val_date1 FROM t_pushdown2 WHERE val_timestamptz = '2011-01-02 12:34:56.123456 Japan BC';
SELECT val_date1 FROM t_pushdown2 WHERE val_timestamptz = '2011-01-02 12:34:56.123456 Japan BC';
EXPLAIN SELECT val_date1 FROM t_pushdown2 WHERE val_timestampltz = '2011-01-02 BC 12:34:56.123456 Japan';
SELECT val_date1 FROM t_pushdown2 WHERE val_timestampltz = '2011-01-02 BC 12:34:56.123456 Japan';

EXPLAIN SELECT * FROM t_pushdown_num WHERE val_binary_float = 'nan'::real AND val_binary_double = 'nan';
SELECT * FROM t_pushdown_num WHERE val_binary_float = 'nan'::real AND val_binary_double = 'nan';
EXPLAIN SELECT * FROM t_pushdown_num WHERE val_binary_float = 'nan' AND val_binary_double = 'nan';
SELECT * FROM t_pushdown_num WHERE val_binary_float = 'nan' AND val_binary_double = 'nan';
EXPLAIN SELECT * FROM t_pushdown_num WHERE val_binary_float = 'infinity'::real AND val_binary_double = 'infinity';
SELECT * FROM t_pushdown_num WHERE val_binary_float = 'infinity'::real AND val_binary_double = 'infinity';
EXPLAIN SELECT * FROM t_pushdown_num WHERE val_binary_float = 'infinity' AND val_binary_double = 'infinity';
SELECT * FROM t_pushdown_num WHERE val_binary_float = 'infinity' AND val_binary_double = 'infinity';
EXPLAIN SELECT * FROM t_pushdown_num WHERE val_binary_float = '-infinity'::real AND val_binary_double = '-infinity';
SELECT * FROM t_pushdown_num WHERE val_binary_float = '-infinity'::real AND val_binary_double = '-infinity';
EXPLAIN SELECT * FROM t_pushdown_num WHERE val_binary_float = '-infinity' AND val_binary_double = '-infinity';
SELECT * FROM t_pushdown_num WHERE val_binary_float = '-infinity' AND val_binary_double = '-infinity';
