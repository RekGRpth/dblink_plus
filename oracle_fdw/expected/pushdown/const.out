\c contrib_regression_utf8
EXPLAIN SELECT id FROM number_numeric1 WHERE val = '1000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000';
                                                                                                                      QUERY PLAN                                                                                                                      
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on number_numeric1  (cost=100.00..100.00 rows=1000 width=4)
   Remote SQL: SELECT "id", NULL FROM "oracle_fdw"."number_numeric1" WHERE ("val" = 1000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000)
(2 rows)

SELECT id FROM number_numeric1 WHERE val = '1000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000';
ERROR:  OCI_ERROR - ORA-01426: 数値オーバーフローが発生しました

EXPLAIN (COSTS FALSE) SELECT * FROM tstz9_tstz WHERE val < '2011-02-03 12:34:56.123457 Asia/Tokyo' AND val >= '2011-02-03 12:34:56.123456 Asia/Tokyo';
                                                                                                                                          QUERY PLAN                                                                                                                                           
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on tstz9_tstz
   Remote SQL: SELECT "id", "val" FROM "oracle_fdw"."timestamptz_9" WHERE (("val" < TO_TIMESTAMP_TZ('2011-02-02 AD 19:34:56.123457 PST8PDT', 'YYYY-MM-DD BC HH24:MI:SS.FF6 TZR')) AND ("val" >= TO_TIMESTAMP_TZ('2011-02-02 AD 19:34:56.123456 PST8PDT', 'YYYY-MM-DD BC HH24:MI:SS.FF6 TZR')))
(2 rows)

SELECT * FROM tstz9_tstz WHERE val >= '2011-02-03 12:34:56.123456 Asia/Tokyo' AND val < '2011-02-03 12:34:56.123457 Asia/Tokyo';
 id | val 
----+-----
(0 rows)

SELECT * FROM tstz9_tstz WHERE val = '2011-02-03 12:34:56.123456 Asia/Tokyo';
 id | val 
----+-----
(0 rows)

SELECT * FROM tstz9_tstz WHERE val = '2011-02-03 12:34:56.123457 Asia/Tokyo';
 id | val 
----+-----
(0 rows)

\q
